administrator/components/com_content/controllers/article.php

```diff
@@ -145,6 +146,47 @@ public function batch($model = null)
 	}
 
 	/**
+	 * Method to generate and store share token.
+	 *
+	 * @return  boolean   True if token successfully stored, false otherwise and internal error is set.
+	 *
+	 * @since   3.7
+	 */
+	public function shareDraft()
+	{
+		$app    = JFactory::getApplication();
+		
+		if (!JSession::checkToken('get'))
+		{
+			$app->enqueueMessage(JText::_('JINVALID_TOKEN'));
+			echo new JResponseJson;
+			$app->close();
+		}
+		
+		$error = false;
+		$message = JText::_('COM_CONTENT_TOKEN_SAVED');
+		
+		try
+		{
+		$title = $this->input->getArray(array('jform' => array('title' => 'string')));
+
+		// Get the model
+		$model = $this->getModel();
+		$return = $model->shareToken($title);
+		JFactory::getApplication()->close();
+		}
+		catch (Exception $e)
+		{
+		$error = true;
+		$message = JText::_('COM_CONTENT_TOKEN_ERROR');
+		echo new JResponseJson($e);
+		}
+		
+		echo new JResponseJson($return, $message, $error);
```

Comment:
false, because if all is good the return will be set to true. The action is not successful unless told it is successful.